src/pad_tail.rs

```diff
@@ -111,30 +95,15 @@ where
             return self.iter.next_back();
         }
 
-        let elements_remaining = self.elements_required - total_consumed;
+        let index_from_back = self.elements_required - self.elements_from_next_back - 1;
         self.elements_from_next_back += 1;
 
-        if self.iter.len() < elements_remaining {
-            Some((self.filler)(
-                self.elements_required - self.elements_from_next_back,
-            ))
+        if index_from_back >= self.iter.len() {
+            Some((self.filler)(index_from_back))
         } else {
             self.iter.next_back()
```

Comment:
> 2\. 'm confused about the need to `assert`, either in this version or the previous version of the code (since they both contain the same logic). self.iter.next_back() is always Some, so I don't know what the assert call does. incrementing self.elements_from_next_back doesn't impact the underlying iterator.

Please add `debug_assert` that the element obtained from `next_back` actually `is_some`. We could leave it out with the argument "garbage in, garbage out", but the `debug_asserts` catches the in-garbage instead of tunneling through wrong results.